### DISCUSSION TITLE
Expand operation not carried out when all component quantities are greater than 1

### DISCUSSION THREAD

**Discussion by ctrevorrow** [Shopify/function-examples#226] [Feedback: Cart Transform API]
When all of a bundle product's component products' quantities are greater than 1, the Expand operation which the Function outputs is not carried out.  However, if one of the component products' quantities is equal to 1, the Expand operation results in the expected cart changes.

For example, this function output results in no changes in the checkout (Line #0 stays as the bundle product and DOES NOT display the components under it):
```
{
  "operations": [
    {
      "expand": {
        "cartLineId": "gid://shopify/CartLine/0",
        "expandedCartItems": [
          {
            "merchandiseId": "gid://shopify/ProductVariant/<component_variant_id_1>",
            "quantity": 3
          },
          {
            "merchandiseId": "gid://shopify/ProductVariant/<component_variant_id_2>",
            "quantity": 2
          },
          {
            "merchandiseId": "gid://shopify/ProductVariant/<component_variant_id_3>",
            "quantity": 4
          }
        ]
      }
    }
  ]
}
```
I can see the operation in the Run details (under Shopify Partners), but no changes are made to the Cart.  

However, this function output results in the expected behavior in the checkout (Line #0 shows the bundle product's name with the components nested under it):
```
{
  "operations": [
    {
      "expand": {
        "cartLineId": "gid://shopify/CartLine/0",
        "expandedCartItems": [
          {
            "merchandiseId": "gid://shopify/ProductVariant/<component_variant_id_1>",
            "quantity": 3
          },
          {
            "merchandiseId": "gid://shopify/ProductVariant/<component_variant_id_2>",
            "quantity": 1
          },
          {
            "merchandiseId": "gid://shopify/ProductVariant/<component_variant_id_3>",
            "quantity": 4
          }
        ]
      }
    }
  ]
}
```

I am fairly certain both examples were working correctly in the first week of April.

**Comment by ctrevorrow** [comment]
I've also tried Shopify's Cart Transform test app and observed the same issue.

Another example, this function output results in Line #0 staying unchanged (not expanded) but Line #1 is expanded and displayed with the components nested under it in the checkout:
```
{
  "operations": [
    {
      "expand": {
        "cartLineId": "gid://shopify/CartLine/0",
        "expandedCartItems": [
          {
            "merchandiseId": "gid://shopify/ProductVariant/<component_variant_id_A>",
            "quantity": 2
          },
          {
            "merchandiseId": "gid://shopify/ProductVariant/<component_variant_id_B>",
            "quantity": 3
          }
        ]
      }
    },
    {
      "expand": {
        "cartLineId": "gid://shopify/CartLine/1",
        "expandedCartItems": [
          {
            "merchandiseId": "gid://shopify/ProductVariant/<component_variant_id_C>",
            "quantity": 2
          },
          {
            "merchandiseId": "gid://shopify/ProductVariant/<component_variant_id_D>",
            "quantity": 1
          },
          {
            "merchandiseId": "gid://shopify/ProductVariant/<component_variant_id_E>",
            "quantity": 3
          }
        ]
      }
    }
  ]
}
```

When going through with a checkout, the order shows up in the admin with the line items like Line #0 still unexpanded.  If a bundle does not have at least one component with a quantity of 1, it is never expanded at any stage.

**Comment by ignacio-chiazzo** [comment]
Hey! Thanks for bringing this up. 

I replied in [the issue ](https://github.com/Shopify/function-examples/issues/234#issuecomment-1526158134).
